### ISSUE TITLE
bug(openapi): Schema Has A Required Property Undefined (2bd608ae) fires as false positive when required property is the sole entry in `properties`

### ISSUE THREAD

**Issue by haroldb1** [Checkmarx/kics#8081]
# bug(openapi): Schema Has A Required Property Undefined (2bd608ae) fires as false positive when required property is the sole entry in `properties`

## Summary

Query `2bd608ae` (`Schema Has A Required Property Undefined (v3)`) produces false positives on OpenAPI 3.0 schemas where the required property **is** defined in `properties`, but is the **only** property (no siblings). It also fails to detect genuine violations where the required property is truly absent but other properties exist (false negatives).

## KICS version

2.1.20

## Affected query

`assets/queries/openAPI/general/schema_required_property_undefined/query.rego`

## Root cause

The Rego predicate used to detect a missing property is:

```rego
requiredProperty := schema.required[_]

all([property | property != requiredProperty; _ := schema.properties[property]])
```

The comprehension collects every property in `schema.properties` whose name is **not** `requiredProperty`. Then `all(list)` is called on the result.

**False positive case** — required property is the only entry in `properties`:

```
schema.required    = ["name"]
schema.properties  = {name: {type: string}}
```

The comprehension filter `property != "name"` excludes the only property, producing `[]`.
`all([])` evaluates to `true` (vacuous truth) → rule fires incorrectly.

**False negative case** — required property is genuinely absent but other properties exist:

```
schema.required    = ["missing"]
schema.properties  = {name: {type: string}}
```

The comprehension produces `["name"]` (a non-empty list of strings).
`all(["name"])` evaluates to `false` (strings are not boolean `true`) → rule does NOT fire, missing a real violation.

## Verified reproduction

```yaml
openapi: "3.0.0"
info:
  title: Repro
  version: "1.0.0"
paths: {}
components:
  schemas:

    # FALSE POSITIVE — fires even though `name` IS defined
    OnlyRequiredProperty:
      type: object
      additionalProperties: false
      required:
        - name
      properties:
        name:
          type: string

    # CORRECT — does not fire (name defined alongside sibling)
    RequiredWithSiblings:
      type: object
      additionalProperties: false
      required:
        - name
      properties:
        name:
          type: string
        description:
          type: string

    # FALSE NEGATIVE — does not fire even though `missing` is absent from properties
    TruePositive:
      type: object
      additionalProperties: false
      required:
        - missing
      properties:
        name:
          type: string
```

Running `kics scan --include-queries 2bd608ae-8a1f-457f-b710-c237883cb313` on the above:
- `OnlyRequiredProperty` → **flagged** (false positive)
- `RequiredWithSiblings` → not flagged (correct)
- `TruePositive` → **not flagged** (false negative — real violation missed)

## Suggested fix

Replace the inverted `all()` predicate with a direct absence check:

```rego
not schema.properties[requiredProperty]
```

This fires if and only if `requiredProperty` has no entry in `properties`, regardless of how many other properties exist.
